R/normalize.R

```diff
@@ -0,0 +1,98 @@
+##' @export
+##' 
+##' @rdname normalize
+normalizeMethods <- function()
+    c("sum", "max", "center.mean",
+      "center.median", "diff.median",
+      "quantiles", "quantiles.robust", "vsn")
+
+
+##' @title Quantitative data normalisation
+##'
+##' @description
+##'
+##' Function to normalise a matrix of quantitative omics data. The
+##' nature of the normalisation is controlled by the `method`
+##' argument, described below.
+##'
+##' @details
+##'
+##' The `method` parameter in `normalize` can be one of `"sum"`,
+##' `"max"`, `"quantiles"`, `"center.mean"`, `"center.median"`,
+##' `"center.median"`, `"quantiles.robust`" or `"vsn"`.  For `"sum"`
+##' and `"max"`, each feature's intensity is divided by the maximum or
+##' the sum of the feature respectively. These two methods are applied
+##' along the features (rows). The `normalizeMethods()` function
+##' returns a vector of available normalisation methods.
+##'
+##' `"center.mean"` and `"center.median"` translate the respective
+##' sample (column) intensities according to the column mean or
+##' median. `"diff.median"` translates all samples (columns) so that
```

Comment:
I think it would be good if you mentioned here that in fact the column mean or median is **subtracted** from each value of that column. I usually normalize the data by dividing the column intensities by the column median / median across all column medians - so, applying this method I was first surprised by its result.

Also, I was at first puzzled by the word *translate* here - maybe something like *centers* (or *shifts*) might be better?

Reply:
Thank you, that's really misleading. I will change this and add the versions that divide in the next PR.

Reply:
Regarding division *vs* subtraction, I'm totally fine with the subtraction - if one wanted to use a division instead it would be pretty straight forward by first log2 transforming the data. So, the subtraction is OK, I would just clarify this in the documentation.

Reply:
I think the best is the have `center.mean` and `center.media` using the subtraction (that's how `scale` works) and add `div.mean` and `div.median`. What do you think?

Reply:
Note that it was possible to set your expected behaviour by using `normalize_matrix(x, "center.mean", FUN = "/")`, but that wasn't clear, so I prefer to add the `div.*` methods.

Reply:
Sure, makes sense.